src/module-elasticsuite-catalog/Search/Request/Product/Aggregation/Provider/FilterableAttributes/Autocomplete/AttributeList.php

```diff
@@ -24,38 +26,101 @@
+     * @param CacheInterface                                                           $cache             Cache
+     * @param SerializerInterface                                                      $serializer        Serializer
      */
     public function __construct(
-        \Magento\Catalog\Model\ResourceModel\Product\Attribute\CollectionFactory $collectionFactory
+        \Magento\Catalog\Model\ResourceModel\Product\Attribute\CollectionFactory $collectionFactory,
+        CacheInterface $cache,
+        SerializerInterface $serializer
     ) {
         $this->collectionFactory = $collectionFactory;
+        $this->cache             = $cache;
+        $this->serializer        = $serializer;
     }
 
     /**
      * {@inheritdoc}
+     *
+     * @SuppressWarnings(PHPMD.ElseExpression)
      */
     public function getList()
     {
-        /** @var $collection \Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection */
-        $collection = $this->collectionFactory->create();
-        $collection->setItemObjectClass(\Magento\Catalog\Model\ResourceModel\Eav\Attribute::class)
-            ->setOrder('position', 'ASC');
+        if (null === $this->attributeList) {
+            $cacheKey       = self::CACHE_KEY;
+            $attributesList = $this->cache->load($cacheKey);
+
+            if (false === $attributesList) {
+                /** @var $collection \Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection */
+                $collection = $this->collectionFactory->create();
+                $collection->setItemObjectClass(\Magento\Catalog\Model\ResourceModel\Eav\Attribute::class)
+                    ->setOrder('position', 'ASC');
+
+                $collection->addSetInfo(true);
+                $collection->addFieldToFilter('additional_table.is_displayed_in_autocomplete', ['eq' => 1]);
+                $collection->setOrder('attribute_id', 'ASC');
 
-        $collection->addSetInfo(true);
-        $collection->addFieldToFilter('additional_table.is_displayed_in_autocomplete', ['eq' => 1]);
-        $collection->setOrder('attribute_id', 'ASC');
+                $collection->load();
 
-        $collection->load();
+                $this->attributeList = $collection->getItems();
+                $cacheData = $this->serializer->serialize($collection->toArray());
 
-        return $collection->getItems();
+                $this->cache->save($cacheData, $cacheKey, $this->getCacheTags());
+            } else {
+                $attributesData = $this->serializer->unserialize($attributesList);
+                $collection     = $this->collectionFactory->create();
+
+                $this->attributeList = [];
+                foreach ($attributesData['items'] ?? [] as $attributeData) {
+                    $item = $collection->getNewEmptyItem();
+                    $item->setData($attributeData);
+                    $this->attributeList[] = $item;
```

Comment:
The only difference with the non-cached version is the loss of attributeId indices.